### ISSUE TITLE
Issue with "Graceful asynchronous programming with Promises": …

### ISSUE THREAD

**Issue by Dewke** [mdn/content#1118]
MDN URL: https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Promises

#### What information was incorrect, unhelpful, or incomplete?
"[...]The callback inside the .then() block (referred to as the executor)"

Code passed to new Promise() is called executor: new Promise(executor). Executor runs automatically when the promise is created.

https://javascript.info/promise-basics:
"The function passed to new Promise is called the executor. When new Promise is created, the executor runs automatically."
https://mithril.js.org/promise.html:
"promise = new Promise(executor)", and:
"executor - A function that determines how the promise will be resolved or rejected"

#### Specific section or headline?
Paragraph number 4 under "Explaining basic promise syntax: A real example" paragraph.

https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Promises#explaining_basic_promise_syntax_a_real_example

#### Did you test this? If so, how?

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/learn/javascript/asynchronous/promises`
* MDN URL: https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Promises
* GitHub URL: https://github.com/mdn/content/blob/master/files/en-us/learn/javascript/asynchronous/promises/index.html
* Report started: 2021-01-09T10:27:59.460Z

</details>

**Comment by chrisdavidmills** [comment]
@Dewke can you explain a bit more clearly what you think is wrong here? The callback inside the `.then()` block is a function passed to a new promise, as calling `.then()` generates a new promise, so it sounds like both statements are correct. The one in our article is a bit more context specific, whereas the one you cite is a bit more generic.

Do you think our article should be more generic too? How would you rephrase it?

**Comment by Dewke** [comment]
I have not seen any other source calling the callback speccified in `.then()` an executor.
I would rather call it a "consumer function".

There are some differences between these two:
The executor code, as in `new Promise(executor)`, accepts two arguments - **resolve** and **reject**. These are the two methods we can use inside the executor code to either resolve or reject the promise.

Consumer code, on the other hand, receives two callbacks handling success and/or failure of the Promise. We do not have the access to **resolve** and **reject** methods.

I believe that mixing these two terms may lead to unnecessary confusion. It seems to me the article should avoid using the term "executor" for consumer functions.

**Comment by chrisdavidmills** [comment]
Thanks for the explanation, @Dewke . I agree that this should be investigated more.

@jadjoubran , you did some work on promises recently. Have you got any thoughts here?

**Comment by jadjoubran** [comment]
Interesting, I will take a look as soon as possible, probably need to read the spec to get a better answer here
Feel free to assign me!

**Comment by chrisdavidmills** [comment]
Cheers @jadjoubran , you're awesome!

**Comment by jadjoubran** [comment]
Indeed, the Promise receives an executor but what the  `.then()` receives should not be called executor.

Reading the [spec](https://tc39.es/ecma262/#sec-promise.prototype.then), keywords such as `onFullfiled` and `thenAction` are used. Also, the existing MDN page for [Promise.prototype.then()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then) uses the simpler term callback.

My personal opinion is to update the document and use `callback` or `fulfilmentCallback`, what are your thoughts? I'll send a PR afterward

**Comment by chrisdavidmills** [comment]
I reckon just "callback" would make sense. Perhaps in the context of an explanation like "callback that is run when the promise is fulfilled (sometimes known as the fullfillment callback)"

**Comment by jadjoubran** [comment]
This can be closed now 😃
